OneOneOne/reciever/message_checks/Check.py

```diff
@@ -0,0 +1,54 @@
+import logging
+
+from builder.pystache_message_builder import PystacheMessageBuilder
+from definitions import TEMPLATE_PATH, XML_PATH
+from utilities.file_utilities import FileUtilities
+
+
+class Check:
+
+    distribution_envelope = "./soap:Body/itk:DistributionEnvelope"
+    soap_body = "./soap:Body"
+    manifest_tag = distribution_envelope + "/itk:header/itk:manifest"
+    basic_success_message = FileUtilities.get_file_string(str(XML_PATH / 'basic_success_response.xml'))
+
+    namespaces = {
+        'soap': 'http://schemas.xmlsoap.org/soap/envelope/',
+        'a': 'http://www.etis.fskab.se/v1.0/ETISws',
+        'wsa': 'http://www.w3.org/2005/08/addressing',
+        'itk': 'urn:nhs-itk:ns:201005'
+    }
+
+    def __init__(self, message):
+        self.message_tree = message
+
+    def check(self):
+        pass
```

Comment:
Cool I take it this is the base class and they'll be specialisation of this function. You might want to use `ABC` Abstract Base Class and tag this function as an `abstractmethod` will make this clear. I've just recently done something similar. Have a look at `nhais-adaptor/adaptor/outgoing/message_adaptor`

Reply:
Updated it to use ABC and I've added some docs to explain it a bit more